### DISCUSSION TITLE
How to tap options on Share sheet

### DISCUSSION THREAD

**Discussion by bilaldbank** [leancodepl/patrol#917] [Q&A]
Is it possible to interact with the Native iOS or Android Share sheet using patrol? Any code sample would be appreciated!

**Comment by bartekpacia** [comment]
Hi! Can you share example screenshots of these share sheets?

On Android, it should be possible by tapping on the text that is visible. For example:

```dart
await $.native.tap(Selector(text: "Telegram")); // share using Telegram
```

On iOS it would be the same, but you'd also have to know the `appId` of the share sheet.

```dart
await $.native.tap(
  Selector(text: "Telegram")); // share using Telegram
  appId: 'com.apple.springboard',
);
```

But I'm not sure if `com.apple.springboard` would work in this case. Trial & error is the only approach here :)

**Comment by bilaldbank** [comment]
![Screenshot_1669956650](https://user-images.githubusercontent.com/119291964/205219209-18a65c2c-8947-4c37-b696-95733c067b62.png)
![Simulator Screen Shot - iPhone 14 Pro Max - 2022-12-02 at 11 27 36](https://user-images.githubusercontent.com/119291964/205229561-6201d7b6-4244-4d38-bd38-0e9a5c4d9306.png)

Will update if I have any success with either Android or iOS

**Comment by bilaldbank** [comment]
@bartekpacia working on Android. For iOS, I might have to add a step for manual interaction with the share sheet for now. Thanks!

**Comment by bartekpacia** [comment]
I'm happy that you got Android working 🚀

As for iOS, try without specifying `appId` at all. And share the results here, to help others :)

**Comment by bilaldbank** [comment]
No success for iOS. Tried with no appId and also with "com.apple.springboard". I have verified that other native commands like `openQuickSettings();` are working correctly when connected to the same simulator.
